### ISSUE TITLE
Greater control over where Assets get uploaded?

### ISSUE THREAD

**Issue by lindseydiloreto** [craftcms/redactor#219]
When adding a native **Assets** field, you get a certain set of options to help control where those assets get stored. But when adding a **Redactor** field, you get a _different_ set of options to help control where images are stored.

Here are your options when editing a Redactor field...

<img width="580" alt="Redactor" src="https://user-images.githubusercontent.com/5309692/80733857-20193480-8ac3-11ea-8e45-e24bff021eb8.png">

And here are your options when editing an Assets field...

<img width="580" alt="Assets" src="https://user-images.githubusercontent.com/5309692/80733915-358e5e80-8ac3-11ea-9da4-44bd89299fd3.png">

The two biggest disparities are boxed out in red above. Without these options for Redactor, I can't ensure that my Redactor fields and regular Assets are being uploaded to the same place. This is even more of an issue if you're trying to keep everything contained to a specific subfolder.

Is there any plan to bring parity between the two field types? In my opinion, Redactor's image handling options should be _identical_ to the native Assets field.

Let me know if you think this is possible! 🙂

**Comment by lindseydiloreto** [comment]
### Use case:

A content builder matrix. I want the Text panel to share an asset folder with the Image panel.

![Ω 2020 04 30-09 40 25@2x](https://user-images.githubusercontent.com/5309692/80736467-0aa60980-8ac7-11ea-8083-aed14f1c17b8.png)

**Comment by wizzywiggs** [comment]
+1 
I have the same exact scenario, so I agree this would be a great way to customize your assets to give users complete control over the assets when you add content without manually picking folders every time. That might get confusing and lead to errors if someone forgets.

**Comment by andris-sevcenko** [comment]
Yeah, this is complicated. The asset field setting isn't just a "point selector to here" setting - it also involves creating folders and files in a temporary volume. And then, after each parent element is saved, Redactor would be scraping its content to find element references, find the corresponding assets and move to a new folder, if required (if the folder path can now be resolved or has changed).

There are a lot of moving parts, which are now contained within the Asset field, logically. Of course, it can be refactored to a trait or something, but first, we'd have to figure out the long-term plan for Redactor. Seeing as how Imperavi have a new WYSIWYG editor now, it's unlikely they'll continue to push Redactor, and, well, there are some things we'd wish they would do differently, anyway.

For now, I'm leaving this issue open, though, as it makes sense to have that.

**Comment by ruru-ink** [comment]
+1
